Tiltfile

```diff
@@ -293,13 +293,60 @@ def enable_provider(name, debug):
             links = links,
         )
 
+        provider_yaml = get_provider(yaml)
+        print("provider_yaml: " + provider_yaml)
+        k8s_yaml(blob(provider_yaml))
+
 def find_manager(yaml):
     manifests = decode_yaml_stream(yaml)
     for m in manifests:
         if m["kind"] == "Deployment":
             return m["metadata"]["name"]
     return ""
 
+# This code reimplements cmd/clusterctl/client/repository/components.InventoryObject
+# It takes all required information from the Namespace resource.
+def get_provider(yaml):
+    manifests = decode_yaml_stream(yaml)
+
+    for m in manifests:
+        if m["kind"] == "Namespace":
+            target_namespace = m["metadata"]["name"]
+            manifest_label = m["metadata"]["labels"]["cluster.x-k8s.io/provider"]
+
+    # Calculate provider type and name from manifest_label.
+    # Examples for manifest labels: "cluster-api", "infrastructure-docker",
+    # "bootstrap-kubeadm", "control-plane-kubeadm"
+    # type is based on the prefix.
+    # name is the manifest label without the provider type prefix.
+    type = "CoreProvider"
+    name = manifest_label
+    if manifest_label.startswith("infrastructure"):
+        type = "InfrastructureProvider"
+        name = manifest_label[len("infrastructure-"):]
+    if manifest_label.startswith("bootstrap"):
+        type = "BootstrapProvider"
+        name = manifest_label[len("bootstrap-"):]
+    if manifest_label.startswith("control-plane"):
+        type = "ControlPlaneProvider"
+        name = manifest_label[len("control-plane-"):]
+
+    yaml = """
```

Comment:
Let's move this up with the other yaml constants, so we keep the code readable

Reply:
Done, it's a bit cleaner now, but on the other side it's harder to match the templating parameters